stockfish/models.py

```diff
@@ -380,6 +388,59 @@ def get_top_moves(self, num_top_moves: int = 5) -> List[dict]:
             self._parameters.update({"MultiPV": old_MultiPV_value})
         return top_moves
 
+    def benchmark(self, **kwargs: Any) -> str:
```

Comment:
Lets rework this function to get parameter `parameters` of type `BenchmarkParameters`

E.g.:
```python
@dataclass
class BenchmarkParameters:
    ttSize: int
    threads: int
    ...
```

Reply:
TypeError: 'BenchmarkParameters' object does not support item assignment
TypeError: 'BenchmarkParameters' object is not iterable

It seems i cannot assign new values once a dataclass is initialized, and i cant iterate over a dataclass. What would be the purpose of using a dataclass over having a dictionary?

Reply:
> What would be the purpose of using a dataclass over having a dictionary?

The purpose is not writing dictionary elements as a comment

```
Kwargs:
  ttSize: int -> Transposition Table size in MB (max 2048)
  threads: int -> Number of search threads that should be used (max 512)
  limit: int -> Limit value of limitType spent for each position (max 10000)
  fenFile: str -> Path to a FEN format file containing positions to bench (path/to/file.fen)
  limitType: str -> Type of the limit used with limit value (depth, perft, nodes, movetime)
  evalType: str -> Evaluation type used (mixed, classical, NNUE)
```

Reply:
The latest commit changes this method to use the dataclass as you suggested. Let me know if its accpetable :)

Reply:
Python 3.6 knows nothing about dataclasses ))

I'll remove it's support now

Reply:
Update your branch with `master` branch pls

Reply:
Done